doc/snmp-schema-addition.md

```diff
@@ -0,0 +1,97 @@
+# SONiC SNMP TABLE schema proposal #
+
+## Scope of the change ##
+
+Currently SNMP configuration is managed from a mix of yaml files and DB based ACLs, we propose to integrate the SMP configs into an SNMP table in the DB.  
+The following document proposes the database Schema in Json format and the list of required changes to the code.
+
+## Current configurtions for SNMP ##
+sonic-buidimage and sonic-snmpagent depend on the following files:
+1. */etc/sonic/snmp.yaml*
+    defines 2 keys:  
+        *snmp_rocommunity:*  
+        *snmp_location:*  
+   this file is consumed in the snmp docker container by */usr/share/sonic/templates/snmpd.conf.j2* and */usr/bin/snmpd-config-updater*.  
+2. */etc/sonic/config_db.json* or */etc/sonic/minigraph.xml* for ACL definitions 'SNMP_ACL'
+   this file is consumed by */usr/bin/snmpd-config-updater*, */usr/share/sonic/templates/alias_map.j2*, */usr/share/sonic/templates/sysDescription.j2*.
+
+In the end this is used to produce */etc/snmp/snmpd.conf*.
+
+## Limits and incentives to change ##
+- This goes against the principle of having the config DB be the central repository of configuration data.
+- Imposes the provisioning of multiple static files.
+- the snmpContact is hardcoded in the */usr/share/sonic/templates/snmpd.conf.j2* file.
+
+## Proposal: integrate the SNMP configuration into config DB ##
+### Proposed Schema ###
+
+```
+{"SNMP": [ {"LOCATION": "LOCATION_STRING",
+            "CONTACT": "CONTACT_STRING",
+            "SNMP_COMMUNITIES": [ {"SNMP_COMMUNITY": "SNMP_COMMUNITY_STRING1", "TYPE":"RO|RW"},
```

Comment:
>SNMP_COMMUNITIES [](start = 13, length = 16)

Currently our ConfigDB has these levels in any database: Table / Key / Field / Value. So we could not implement too many levels in json.
ref: [https://github.com/Azure/SONiC/blob/2a3be770db988c3d089ac54d6ee3eb9bb77ae621/doc/config_db.json](https://github.com/Azure/SONiC/blob/2a3be770db988c3d089ac54d6ee3eb9bb77ae621/doc/config_db.json)
All the above level/below levels are key-value pairs, so we don't use json array.
Suggest you treat SNMP_COMMUNITY_STRING1 as key, and other field/values pairs as its value.